provider/pkg/gen/schema.go

```diff
@@ -1386,6 +1396,51 @@ func (g *packageGenerator) genTypeSpec(typeName, propName string, prop *discover
 			Type: "object",
 			Ref:  referencedTypeName,
 		}, nil
+	case prop.Type == "object" && prop.AdditionalProperties != nil:
+		// The prop is a map with a string key and a simple value
+		// if it doesn't have a ref.
+		if prop.AdditionalProperties.Ref == "" {
+			switch prop.AdditionalProperties.Type {
+			case "any":
+				return &schema.TypeSpec{
+					Type:                 "object",
+					AdditionalProperties: &schema.TypeSpec{Ref: "pulumi.json#/Any"},
+				}, nil
+			case "array":
+				// An array type for AdditionalProperties doesn't make sense
+				// since it's already a map and multiple entries are
+				// allowed by definition. So generate a type
+				// based on the type of the "items" in the array.
```

Comment:
> a `map[string][]T` is a perfectly valid type (eg. you might have a map from team name to employees list).

Yes. But I wasn't referring to that :) My confusion was about `type: array` and what that is referring to. I guess I didn't realize that the `type: array` here is indicating a list of values for each key. If that is the case, then yeah what I have here is definitely wrong. I'll change this.

Reply:
I've fixed this now.

Reply:
By the way, the reason why the array type didn't make sense to me in the first place in this context is, if you look at the UI for creating a budget, you can only have a single value for a key: 
![image](https://github.com/pulumi/pulumi-google-native/assets/1466314/2d9dec35-14cc-4e50-9c43-cee7aaa18af4)

Having said that I agree that this can't be generalized like how I did at first because there could be a `<string, any[]>` elsewhere in the future that this would break.